site/docs/patterns/forms.mdx

```diff
@@ -0,0 +1,839 @@
+    </p>
+    <ul>
+        <li>Use smaller spacing between controls that belong together and larger spacing between groups of controls</li>
+    </ul>
+    <Image src="../../static/patterns/form/form-layout-spacing@2x.png" alt="Use spacing for grouping form controls" style="max-width:432px;" viewable /> 
+</div>
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <p>
+        <IconCheck style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle', color: 'var(--color-success)' }} size="s"/>
+        <strong>Align form controls to <a href="/guidelines/grid">column grid</a>.</strong>
+    </p>
+    <Image src="../../static/patterns/form/form-layout-grid@2x.png" alt="Align fields to column grid" style="max-width:432px;" viewable /> 
+</div>
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <p>
+        <IconCheck style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle', color: 'var(--color-success)' }} size="s"/>
+        <strong>Adjust field width according to preferred input.</strong>
+    </p>
+    <Image src="../../static/patterns/form/form-layout-width@2x.png" alt="Adjust field width according to preferred input" style="max-width:432px;" viewable /> 
+</div>
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <p>
+        <IconCheck style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle', color: 'var(--color-success)' }} size="s"/>
+        <strong>Align form wide action buttons at the end of the form, aligned left in order of importance</strong>
+    </p>
+    <ol>
+        <li>Submit (primary action)</li>
+        <li>Secondary actions</li>
+    </ol>
+    <Image src="../../static/patterns/form/form-layout-actions-form@2x.png" alt="Align form wide action buttons at the end of the form" style="max-width:432px;" viewable /> 
+</div>
+
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+        <p>
+            <IconCheck style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle', color: 'var(--color-success)' }} size="s"/>
+            <strong>Align supplementary action buttons related to singe form control or fieldset after or bellow the control</strong>
+        </p>
+        <Image src="../../static/patterns/form/form-layout-actions-field@2x.png" alt="Supplementary action" style="max-width:432px;" viewable />
+</div>
+
+---
+
+### 2. Grouping related controls
+**Form controls that perform a common task or share a common purpose *should* be grouped together** within a 'fieldset' element and labelled with a 'legend' element, which describes the purpose of the form control group. 
```

Comment:
> fieldset ... legend

Nämä voisi kääriä kooditageihin.

Reply:
Piti laittaa, muuta jostain syystä kooditagit ei näy inlinena, kun ne laittaa html-tagien sisään… jonkun pitäis oikeesti kattoa vielä tää koodi järkevämmäksi.